src/ai/defaultai.cc

```diff
@@ -5662,18 +5632,16 @@ BuildingNecessity DefaultAI::check_building_necessity(BuildingObserver& bo,
 			}
 		} else if (bo.type == BuildingObserver::Type::kMine) {
 			bo.primary_priority = bo.max_needed_preciousness;
-			if ((mines_per_type[bo.mines].in_construction + mines_per_type[bo.mines].finished) == 0 &&
+			if (mines_per_type[bo.mines].total_count() == 0 &&
 			    site_needed_for_economy != BasicEconomyBuildingStatus::kDiscouraged) {
 				// unless a mine is prohibited, we want to have at least one of the kind
 				bo.max_needed_preciousness = bo.max_preciousness;
 				return BuildingNecessity::kNeeded;
-			} else if (((mines_per_type[bo.mines].in_construction +
-			             mines_per_type[bo.mines].finished) == 1) &&
-			           bo.is(BuildingAttribute::kBuildingMatProducer) &&
+			} else if (mines_per_type[bo.mines].finished == mines_per_type[bo.mines].total_count() &&
+			           bo.current_stats >
+			              (85 + std::abs(management_data.get_military_number_at(173)) / 10) &&
```

Comment:
you remove number at 129 (few lines below) and add 173. Intentional?

Reply:
no that was 2 single steps. First in the deletion I was not aware that I freed a number. So I'll reuse 129 for this.  
Always good to have a reviewer